### DISCUSSION TITLE
Integration tests with react testing library and server actions

### DISCUSSION THREAD

**Discussion by Magefo** [vercel/next.js#56304] [Help]
### Summary

I would like to add some integration tests to a form component that utilizes `experimental_useFormState` and `experimental_useFormStatus`. My testing stack comprises `jest` and `@testing-library/react`, but I have been unable to find an effective method to accomplish this. I am wondering if anyone has successfully managed this and could provide some examples. Thank you. 😄

**Comment by vijote** [comment]
Hi, I don't have experience with integration tests, but I unit tested forms and server actions using several mocks and React 18.3(canary).

With React 18.2 I wasn't able to make it
